src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt

```diff
@@ -0,0 +1,278 @@
+        return ecsService
+    }
+
+    /**
+     * Constructs a [RunTaskRequest] with Task Overrides using the passed in parameters. [TaskOverrides][TaskOverride]
+     * are constructed from the [overrides] and applied to the ECS task definition [taskDefinition].
+     * ECS tasks will be run on EC2 instances.
+     *
+     * @param taskDefinition  The family and revision (family:revision) or full ARN of the task definition to run.
+     * If a revision is not specified, the latest ACTIVE revision is used.
+     */
+    fun buildEcsTask(ecsClusterName: String, taskDefinition: String, taskRoleArn: String, overrides: Collection<ContainerOverride>): RunTaskRequest {
+        val taskOverride = TaskOverride.builder()
+                .containerOverrides(overrides)
+                .taskRoleArn(taskRoleArn)
+                .build()
+
+        return RunTaskRequest.builder()
+                .cluster(ecsClusterName)
+                .launchType("EC2")
+                .overrides(taskOverride)
+                .taskDefinition(taskDefinition)
+                .build()
+    }
+
+    /**
+     * Runs the specified [RunTaskRequest]
+     */
+    fun runEcsTask(taskRequest: RunTaskRequest) {
+        val task = awsClients.ecsClient.runTask(taskRequest).tasks()[0]
+        logger.info("ECS tasks run",
+                "instance_arns" to task.containerInstanceArn(),
+                "task_groups" to task.group(),
+                "cluster_arn" to task.clusterArn(),
+                "cpus" to task.cpu(),
+                "memory" to task.memory(),
+                "platform_version" to task.platformVersion(),
+                "started_at" to task.startedAt().toString())
+    }
+
+    /**
+     * Creates an IAM [Policy] from the name and document provided. [policyDocument] should be in JSON format
+     * as per the AWS standards for documents.
+     */
+    fun createIamPolicy(policyName: String, policyDocument: String): Policy {
+        val policy = awsClients.iamClient.createPolicy(
+                CreatePolicyRequest.builder()
+                        .policyDocument(policyDocument)
+                        .policyName(policyName).build())
+                .policy()
+        logger.info("Created iam policy",
+                "policy_arn" to policy.arn(),
+                "policy_path" to policy.path(),
+                "policy_name" to policy.policyName(),
+                "policy_id" to policy.policyId(),
+                "created_date" to policy.createDate().toString())
+        return policy
+    }
+
+    /**
+     * Creates an IAM [Role] from the name and role assumption document provided. [assumeRolePolicy] should be
+     * in JSON format as per the AWS standards for documents.
+     */
+    fun createIamRole(roleName: String, assumeRolePolicy: String): Role {
+        val role = awsClients.iamClient.createRole(
+                CreateRoleRequest.builder()
+                        .assumeRolePolicyDocument(roleName)
```

Comment:
This should probably be 
```suggestion
                        .assumeRolePolicyDocument(assumeRolePolicy)
```

Reply:
Good catch! Will update